magento/pwa-studio
# [PWA-1016] Refactor Add To Cart to use generic AddProductsToCart mutation

## Description

In Magento 2.4.1 there is a new AddProductsToCart mutation which is more generic and type agnostic than the older, type specific mutations such as "AddConfigurableProductsToCart".

1) Investigate whether we can replace the Simple and Configurable add mutations with the generic one. If so, please do the work as part of this ticket.

2) Investigate whether it can support future types such as Bundled/Virtual/etc. If so, we don't need the "Supported Product Types" logic anymore, so please remove that too.

Links:

https://i.gyazo.com/5e19a2e43a1b46654d053197a280c1ca.png
https://github.com/magento/pwa-studio/pull/2769#issuecomment-708097397
https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js#L237-L262

## Related Issue





* [[PWA-1016](https://jira.corp.magento.com/browse/PWA-1016)] Refactor Add To Cart to use generic AddProductsToCart mutation

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


1. Verify guest shoppers can add simple and configurable products to their cart
2. Verify logged in shoppers can add simple and configurable products to their cart

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
